controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1112,21 +1121,45 @@ private long findSegmentSplitsMerges(List<StreamSegmentRecord> referenceSegments
         });
     }
 
-    private CompletableFuture<Version> sealActiveTx(final int epoch, final UUID txId, final boolean commit,
+    private CompletableFuture<Version> sealActiveTx(final UUID txId, final boolean commit,
                                                     final ActiveTxnRecord previous,
+                                                    final String writerId, 
+                                                    long time, 
                                                     final Version version) {
-        final ActiveTxnRecord updated = new ActiveTxnRecord(previous.getTxCreationTimestamp(),
-                previous.getLeaseExpiryTime(),
-                previous.getMaxExecutionExpiryTime(),
-                commit ? TxnStatus.COMMITTING : TxnStatus.ABORTING);
-        final Data data = new Data(updated.toBytes(), version);
-        return updateActiveTx(epoch, txId, data);
+        CompletableFuture<ActiveTxnRecord> future;
+        if (commit) {
+            // if request is for commit --> add a new entry to orderer. 
+            // update record
+            if (!previous.getTxnStatus().equals(TxnStatus.COMMITTING)) {
+                future = addTxnToCommitOrder(txId)
+                        .thenApply(position -> {
+                            return new ActiveTxnRecord(previous.getTxCreationTimestamp(),
+                                    previous.getLeaseExpiryTime(),
+                                    previous.getMaxExecutionExpiryTime(),
+                                    TxnStatus.COMMITTING,
+                                    writerId,
+                                    time,
+                                    position);
+                        });
+            } else {
+                future = CompletableFuture.completedFuture(previous);
+            }
+        } else {
+            future = CompletableFuture.completedFuture(new ActiveTxnRecord(previous.getTxCreationTimestamp(),
+                    previous.getLeaseExpiryTime(),
+                    previous.getMaxExecutionExpiryTime(),
+                    TxnStatus.ABORTING,
+                    writerId,
+                    time, Long.MIN_VALUE));
+        }
+        return future.thenCompose(updated -> {
+            final Data data = new Data(updated.toBytes(), version);
+            return updateActiveTx(txId, data);  
+        });
     }
 
-    @Override
+    @VisibleForTesting
```

Comment:
Override?

Reply:
same as above comment about removing from the interface.